### PR TITLE
Only shut down pillows/celery when migrations exist

### PR DESCRIPTION
**Pull request by benrudolph**
@czue forgot i worked on this until i saw your email. also @gcapalbo. not in love with this solution as it seems pretty brittle, but it's not the worst if we upgrade django versions and don't realize for a few deploys that pillows/celery isn't being stopped since we still will always run migrate

cc: @esoergel

### CODE REVIEW

**Comment by czue** [comment]
nice

**Comment by snopoke** [comment]
Are you sure this works? I tried this before bit there was an issue with the `sync_finish_couchdb` command http://manage.dimagi.com/default.asp?144286

**Comment by czue** [comment]
@snopoke I think this change doesn't impact migrations running on every deploy - just only stops pillows/celery if there are expected postgres migrations.

**Comment by snopoke** [comment]
:+1:

File: fab/fabfile.py [hunk 1]
```diff
@@ -948,6 +950,19 @@ def _migrate():
...
+def _migrations_exist():
+    """
+    Check if there exists database migrations to run
+    """
+    _require_target()
+    with cd(env.code_root):
+        n_migrations = int(sudo(
+            '%(virtualenv_root)s/bin/python manage.py migrate --list | grep "\[ ]" | wc -l' % env)
+        )
+        return n_migrations > 0
```
Comment (esoergel) [inline fab/fabfile.py]: Not quite sure how to implement in this context, but FYI, here's how django checks for un-run migrations:
https://github.com/django/django/blob/master/django/core/management/commands/runserver.py#L157-L178
```python
def _migrations_exist():
    executor = MigrationExecutor(connections[DEFAULT_DB_ALIAS])
    return executor.migration_plan(executor.loader.graph.leaf_nodes())
```

Comment (benrudolph) [inline fab/fabfile.py]: interesting thanks. i think i'd have to create a separate management command in order to do this and then maybe key off the exit code. will put it in the back log to do

Comment (esoergel) [inline fab/fabfile.py]: Probably not necessary to implement.  I was curious if there was a more precise way to check it, so I tracked this down.  Figured it was worth mentioning.

**Pull request merged**